de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoice/model/validator/C_Invoice.java

```diff
@@ -188,16 +188,18 @@ public void allocateInvoiceAgainstCreditMemo(final I_C_Invoice creditMemo)
 
 		final I_C_Invoice parentInvoice = InterfaceWrapperHelper.create(creditMemo.getRef_Invoice(), I_C_Invoice.class);
```

Comment:
will `creditMemo.getRef_Invoice()` always return null if `creditMemo.getRef_Invoice_ID()` is null? i mean, with POWrapper, and POJOWrapper? Tbh I'm not 100% sure. If you aren't either, please check for `creditMemo.getRef_Invoice_ID() > 0` instead, before getting the parent invoice.

Otherwise, `parentInvoice` might not be `null` but an "empty" object without values inside.

Reply:
done